### PR TITLE
Add validation on expire date

### PR DESCRIPTION
**Pull request by pantelm**

### CODE REVIEW

**Comment by pantelm** [comment]
Travis fail cause this pull request need [this pull request](https://github.com/Payum/Payum/pull/346) to be merged

File: Tests/Functional/Form/Type/CreditCardTypeTest.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ public function shouldSubmitDataCorrectly()
         $this->assertEquals('John Doe', $card->getHolder());
         $this->assertEquals('4111111111111111', $card->getNumber());
         $this->assertEquals('123', $card->getSecurityCode());
-        $this->assertEquals('2020-10-01', $card->getExpireAt()->format('Y-m-d'));
+        $this->assertEquals('2020-10-31', $card->getExpireAt()->format('Y-m-d'));
```
Comment (makasim) [inline Tests/Functional/Form/Type/CreditCardTypeTest.php]: Could you add a test where credit card constraint fails.

**Comment by pantelm** [comment]
As I use buildviolation in the validator, the test will not pass on symfony <= 2.5

**Comment by makasim** [comment]
@pantelm  We can skip it for old version, no problems here

**Comment by pantelm** [comment]
Ok, but I don't know how to skip depending on version

**Comment by makasim** [comment]
Like this https://github.com/Payum/PayumBundle/pull/133/files#diff-d83ec829930974b5570064c53f84a0d2L215. and inside the `if` statement you can call `$this->markTestSkipped('Works only for symfony >= 2.5');`

**Comment by pantelm** [comment]
Maybe I have to change the validator so it can work on previous version because with the actual use, the validator will send an error.

**Comment by makasim** [comment]
It would be good, or we can add this validator only for symfony 2.5 and later.

**Comment by pantelm** [comment]
@makasim can you launch the test on this RP cause I don't have anything more to commit.

**Comment by makasim** [comment]
merged to master.

**Pull request closed without merging**